#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_gcc43_build_fixes.dpatch by Daniel Kobras <kobras@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add various includes to account for cleanup in g++ 4.3 standard
## DP: headers. Extended version of original patch by Martin Michlmayr.

@DPATCH@
diff -urNad muse~/muse/ctrl.cpp muse/muse/ctrl.cpp
--- muse~/muse/ctrl.cpp	2003-11-28 22:19:07.000000000 +0100
+++ muse/muse/ctrl.cpp	2008-03-09 14:46:01.000000000 +0100
@@ -9,6 +9,7 @@
 //=========================================================
 
 
+#include <cstdlib>
 #include "globals.h"
 #include "ctrl.h"
 #include "xml.h"
diff -urNad muse~/muse/evdata.h muse/muse/evdata.h
--- muse~/muse/evdata.h	2003-10-31 18:55:16.000000000 +0100
+++ muse/muse/evdata.h	2008-03-09 14:46:22.000000000 +0100
@@ -9,6 +9,7 @@
 #ifndef __EVDATA_H__
 #define __EVDATA_H__
 
+#include <cstring>
 #include <memory.h>
 
 //---------------------------------------------------------
diff -urNad muse~/muse/help.cpp muse/muse/help.cpp
--- muse~/muse/help.cpp	2005-03-29 18:37:34.000000000 +0200
+++ muse/muse/help.cpp	2008-03-09 14:46:01.000000000 +0100
@@ -6,6 +6,7 @@
 //  (C) Copyright 1999/2000 Werner Schweer (ws@seh.de)
 //=========================================================
 
+#include <cstdlib>
 #include <unistd.h>
 #include <qmessagebox.h>
 #include <qprocess.h>
diff -urNad muse~/muse/memory.h muse/muse/memory.h
--- muse~/muse/memory.h	2004-06-07 23:46:08.000000000 +0200
+++ muse/muse/memory.h	2008-03-09 14:46:01.000000000 +0100
@@ -9,6 +9,7 @@
 #ifndef __MEMORY_H__
 #define __MEMORY_H__
 
+#include <cstdlib>
 #include <stdio.h>
 #include <map>
 
diff -urNad muse~/muse/mixer/panknob.cpp muse/muse/mixer/panknob.cpp
--- muse~/muse/mixer/panknob.cpp	2004-01-23 09:41:38.000000000 +0100
+++ muse/muse/mixer/panknob.cpp	2008-03-09 14:46:46.000000000 +0100
@@ -6,6 +6,7 @@
 //  (C) Copyright 2000 Werner Schweer (ws@seh.de)
 //=========================================================
 
+#include <cstring>
 #include "../audio.h"
 #include "panknob.h"
 
diff -urNad muse~/muse/thread.cpp muse/muse/thread.cpp
--- muse~/muse/thread.cpp	2005-11-07 19:45:19.000000000 +0100
+++ muse/muse/thread.cpp	2008-03-09 14:46:01.000000000 +0100
@@ -8,6 +8,7 @@
 
 #include "thread.h"
 #include <stdio.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <sys/mman.h>
 #include <sys/poll.h>
diff -urNad muse~/muse/widgets/poslabel.cpp muse/muse/widgets/poslabel.cpp
--- muse~/muse/widgets/poslabel.cpp	2004-01-11 19:55:37.000000000 +0100
+++ muse/muse/widgets/poslabel.cpp	2008-03-09 14:46:01.000000000 +0100
@@ -6,6 +6,7 @@
 //=========================================================
 
 #include <cmath>
+#include <cstdlib>
 #include <qapplication.h>
 #include <qstyle.h>
 
diff -urNad muse~/synti/deicsonze/deicsonze.cpp muse/synti/deicsonze/deicsonze.cpp
--- muse~/synti/deicsonze/deicsonze.cpp	2004-12-12 21:59:35.000000000 +0100
+++ muse/synti/deicsonze/deicsonze.cpp	2008-03-09 14:46:01.000000000 +0100
@@ -27,6 +27,8 @@
 // 02111-1307, USA or point your web browser to http://www.gnu.org.
 //===========================================================================
 
+#include <cstdlib>
+#include <climits>
 #include <cmath>
 #include <list>
 
diff -urNad muse~/synti/simpledrums/ssplugin.cpp muse/synti/simpledrums/ssplugin.cpp
--- muse~/synti/simpledrums/ssplugin.cpp	2005-05-07 23:24:37.000000000 +0200
+++ muse/synti/simpledrums/ssplugin.cpp	2008-03-09 14:46:01.000000000 +0100
@@ -10,6 +10,7 @@
 //
 //
 
+#include <cstdlib>
 #include <dlfcn.h>
 #include "ssplugin.h"
 #include "common.h"
